crossfire/__init__.py

```diff
@@ -51,16 +57,24 @@ def occurrences(
     )
 
 
+def _make_flatten(data, keys):
+    for item in data:
+        for key in keys:
+            item.update({f"{key}_{k}": v for k, v in item.get(key).items()})
+            item.pop(key)
+    return data
+
+
 def flatten(data, nested_columns=None):
     nested_columns = set(nested_columns or NESTED_COLUMNS)
     if not nested_columns.issubset(NESTED_COLUMNS):
         raise NestedColumnError(nested_columns)
-    if not data:
-        return data
-    if isinstance(data, list):
-        keys = set(data[0].keys()) & nested_columns
-        for item in data:
-            for key in keys:
-                item.update({f"{key}_{k}": v for k, v in item.get(key).items()})
-                item.pop(key)
+    if isinstance(data, DataFrame) and not data.empty:
```

Comment:
What is `DataFrame` here if we end up in the `pass` from line 12?

Reply:
Why don't you check [pur `parser.py`](https://github.com/FelipeSBarros/crossfire/blob/771a750a96de9398d28b44790618b1d85ed0f41c/crossfire/parser.py)?

Reply:
Actually, yes... the try except came from there. But now I can see that I am not considering the "flag" `HAS_PANDAS`.
I will consider using it in the following commit.